src/eset.c

```diff
@@ -204,8 +231,13 @@ eset_first_fit(eset_t *eset, size_t size, bool exact_only,
 		if ((sz_pind2sz(i) >> lg_max_fit) > size) {
 			break;
 		}
-		if (ret == NULL || edata_snad_comp(edata, ret) < 0) {
+		if (ret == NULL || edata_cmp_summary_comp(
+		    eset->bins[i].heap_min, ret_summ) < 0) {
```

Comment:
nit: add a getter function for heap_min, and within the getter do an assertion of heap_min == heap_first

Reply:
I'm not 100% sure I'm parsing this correctly, but I added `assert(ret == NULL || edata_cmp_summary_comp(ret_summ, edata_cmp_summary_get(edata)) == 0);`; is that what you mean? (That seems like a good assert anyways).

Reply:
Looks good. That does what I wanted.